core/storage/blog/gae_models.py

```diff
@@ -0,0 +1,335 @@
+        model have already been exported as a part of the BlogPostModel.
+        """
+        return dict(super(cls, cls).get_export_policy(), **{
+            'author_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'title': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'summary': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'url_fragment': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'tags': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'thumbnail_filename': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'published_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'last_updated': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+        })
+
+    @classmethod
+    def create(cls, blog_id):
+        """Creates a new BlogPostSummaryModel entry.
+
+        Args:
+            blog_id: str. Blog ID of the newly-created blog.
+
+        Returns:
+            BlogPostSummaryModel. The newly created BlogPostSummaryModel
+            instance.
+
+        Raises:
+            Exception. A blog summary model with the given blog ID exists
+                already.
+        """
+        if cls.get_by_id(blog_id):
+            raise Exception(
+                'Blog ID conflict on creating new blog summary model.')
+        return cls(id=blog_id)
+
+    @classmethod
+    def get_blog_post_summary_models(cls, blog_ids):
+        """Returns a list of BlogPostSummaryModels for blogs created by the
+        user.
+
+        Args:
+            blog_ids: list(str). Contains blog ids for which the blog post
+                summary models are to be fetched.
+
+        Returns:
+            list(BlogPostSummaryModel). The list of BlogPostSummaryModel
+            created by the given user.
+        """
+        return cls.get_multi(blog_ids)
+
+
+class BlogPostRightsModel(base_models.BaseModel):
+    """Storage model for rights related to a blog.
+
+    The id of each instance is the blog_id.
+    """
+
+    # The user_id of the blog editors of this blog.
+    editor_ids = datastore_services.StringProperty(indexed=True, repeated=True)
+
+    # Whether this blog is published or not.
+    # False if blog is a draft, True if published.
+    blog_is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model contains data to pseudonymize or delete corresponding
+        to a user: editor_ids field.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether BlogPostRightsModel references user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.editor_ids == user_id
```

Comment:
nit pick: maybe add a comment explaining that this checks that _any_ of the IDs are equal to the `user_id` (I had to look it up myself to be sure, but readers shouldn't be forced to do that).

Reply:
Added.